### PR TITLE
Implement sum banks in refl gui preview tab

### PR DESCRIPTION
**Pull request by gemmaguest**
**Description of work.**

This PR implements summing across banks in the preview tab of the ISIS Reflectometry GUI. When the user changes the selection on the instrument view, the algorithm `ReflectometryISISSumBanks` is called with that selection and the result stored in the model. The presenter is notified when algorithm execution is completed. This is WIP so it currently does nothing further at this point other than log a debug message.

**To test:**

- Load this workspace: [INTER45455_inst.zip](https://github.com/mantidproject/mantid/files/7504266/INTER45455_inst.zip)
- Set the output level to `debug` in the log.
- Open the ISIS Reflectometry GUI
- Select the Preview tab.
- Type in `INTER45455` and click Load. The instrument viewer plot should display the data.
- Click the rectangle-select button and select a region of interest.
- Whenever the ROI is created or changed, you should see the text `Sum banks completed` in the log.

Refs #32772

*This does not require release notes* because **this is part of prototype work on a tab that is disabled in release**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: buildconfig/CMake/CppCheck_Suppressions.txt.in [hunk 1]
```diff
@@ -859,7 +859,7 @@ unreadVariable:${CMAKE_SOURCE_DIR}/qt/scientific_interfaces/Muon/ALCDataLoadingV
 useStlAlgorithm:${CMAKE_SOURCE_DIR}/qt/scientific_interfaces/Muon/ALCInterface.cpp:232
 useInitializationList:${CMAKE_SOURCE_DIR}/qt/scientific_interfaces/ISISReflectometry/Common/OptionDefaults.cpp:18
 useStlAlgorithm:${CMAKE_SOURCE_DIR}/qt/scientific_interfaces/ISISReflectometry/Common/Clipboard.cpp:91
-passedByValue:${CMAKE_SOURCE_DIR}/qt/scientific_interfaces/ISISReflectometry/Reduction/PreviewRow.h:18
+passedByValue:${CMAKE_SOURCE_DIR}/qt/scientific_interfaces/ISISReflectometry/Reduction/PreviewRow.h:19
```
Comment (ConorMFinn) [inline buildconfig/CMake/CppCheck_Suppressions.txt.in:862]: Always worth asking jic, are these trivial to just fix?

File: qt/scientific_interfaces/ISISReflectometry/GUI/Batch/SumBanksAlgorithm.cpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+}
+} // namespace
+
+namespace MantidQt::CustomInterfaces::ISISReflectometry::SumBanks {
+
+/** Create a configured algorithm for summing banks. The algorithm
+ * properties are set from the reduction configuration model and the
+ * given row.
+ * @param model : the reduction configuration model
+ * @param row : the row from the preview tab
```
Comment (ConorMFinn) [inline qt/scientific_interfaces/ISISReflectometry/GUI/Batch/SumBanksAlgorithm.cpp:43]: Could we add the `alg` param to the docstring for completeness?

**Review by cailafinn — CHANGES_REQUESTED** [review]
Looking good, just a couple of code notes:

If a workspace is loaded into the tab, then deleted from the ADS, then attempt to load it into the tab (which will fail), the IV keeps showing the instrument but sum_banks fails as there is no input workspace. Does this need resolving for this PR?

File: buildconfig/CMake/CppCheck_Suppressions.txt.in (hunk 1 above)
Comment (gemmaguest) [reply buildconfig/CMake/CppCheck_Suppressions.txt.in:862]: I don't think so - we want to pass by value because we're doing `std::move`

**Comment by gemmaguest** [comment]
> If a workspace is loaded into the tab, then deleted from the ADS, then attempt to load it into the tab (which will fail), the IV keeps showing the instrument but sum_banks fails as there is no input workspace. Does this need resolving for this PR?

Good spot. This is crashing for me - it's actually a problem on `main`, too. Let's fix that in a separate PR. I've raised issue #32961 to look at this.

**Review by cailafinn — APPROVED** [review]
Looks good to me now and functions as described 👍🏻

**Comment by gemmaguest** [comment]
@mantidproject/gatekeepers Could we get this merged? It's blocking other work.

**Pull request merged**